nomad/leader.go

```diff
@@ -1327,3 +1333,19 @@ func (s *Server) getOrCreateSchedulerConfig() *structs.SchedulerConfiguration {
 
 	return config
 }
+
+func (s *Server) generateClusterID() (string, error) {
+	if !ServersMeetMinimumVersion(s.Members(), minClusterIDVersion, false) {
+		s.logger.Named("core").Warn("cannot initialize cluster ID until all servers are above minimum version", "min_version", minClusterIDVersion)
+		return "", errors.New("cluster ID cannot be created until all servers are above minimum version")
```

Comment:
include minimum version in error message

Also seems like we'll need to be able to detect and ignore this error vs other errors? The cluster will have to continue working normally when this error is returned (even if some features are disabled).

This can be done in the followup PR that uses this functionality though.

Reply:
Yeah I'm punting on exporting error types until it's clear where they're going to be needed